pkg/policy/updaterequest.go

```diff
@@ -7,71 +7,122 @@ import (
 	"github.com/gardener/controller-manager-library/pkg/logger"
 	kyverno "github.com/kyverno/kyverno/api/kyverno/v1"
 	urkyverno "github.com/kyverno/kyverno/api/kyverno/v1beta1"
+	common "github.com/kyverno/kyverno/pkg/background/common"
 	"github.com/kyverno/kyverno/pkg/config"
+	"github.com/kyverno/kyverno/pkg/engine"
+	"github.com/kyverno/kyverno/pkg/engine/response"
+	engineutils "github.com/kyverno/kyverno/pkg/engine/utils"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-func (pc *PolicyController) updateUR(policyKey string, policy kyverno.PolicyInterface) {
+func (pc *PolicyController) updateUR(policyKey string, policy kyverno.PolicyInterface) error {
 	logger := pc.log.WithName("updateUR").WithName(policyKey)
 
-	// TODO: add check for genExisting
-	if !policy.GetSpec().MutateExistingOnPolicyUpdate {
-		logger.V(4).Info("skip policy application on policy event", "policyKey", policyKey, "mutateExiting", policy.GetSpec().MutateExistingOnPolicyUpdate)
-		return
+	if !policy.GetSpec().MutateExistingOnPolicyUpdate && !policy.GetSpec().IsGenerateExistingOnPolicyUpdate() {
+		logger.V(4).Info("skip policy application on policy event", "policyKey", policyKey, "mutateExiting", policy.GetSpec().MutateExistingOnPolicyUpdate, "generateExisting", policy.GetSpec().IsGenerateExistingOnPolicyUpdate())
+		return nil
 	}
 
 	logger.Info("update URs on policy event")
 
+	var errors []error
 	mutateURs := pc.listMutateURs(policyKey, nil)
 	generateURs := pc.listGenerateURs(policyKey, nil)
 	updateUR(pc.kyvernoClient, policyKey, append(mutateURs, generateURs...), pc.log.WithName("updateUR"))
 
 	for _, rule := range policy.GetSpec().Rules {
 		var ruleType urkyverno.RequestType
+
 		if rule.IsMutateExisting() {
 			ruleType = urkyverno.Mutate
-		} else {
-			// TODO: assign generate ruleType
-			continue
-		}
 
-		triggers := getTriggers(rule)
-		for _, trigger := range triggers {
-			var urs []*urkyverno.UpdateRequest
-			if ruleType == urkyverno.Mutate {
-				urs = pc.listMutateURs(policyKey, trigger)
-			} else {
-				urs = pc.listGenerateURs(policyKey, trigger)
+			triggers := generateTriggers(pc.client, rule, pc.log)
+			for _, trigger := range triggers {
+				murs := pc.listMutateURs(policyKey, trigger)
+
+				logger.V(4).Info("UR was created", "rule", rule.Name, "rule type", ruleType, "trigger", trigger.GetNamespace()+trigger.GetName())
```

Comment:
Shouldn't we move this log message inside `if murs != nil {}` check?

Reply:
done